src/infrastructure/validation/logger/FileLogger.java

```diff
@@ -0,0 +1,229 @@
+package infrastructure.validation.logger;
+
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.time.LocalDateTime;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.io.IOException;
+
+/**
+ * FileLogger class that is part of loggerManager,
+ * and used for writing log entries to a log file
+ * Implements ILogger interface
+ * 
+ * @author Navaneeth M Nambiar
+ */
+public class FileLogger implements ILogger {
+
+	/** stores the format specified for the time-stamp */
+	private DateTimeFormatter timeStampFormat;
+		
+	/** string that holds the location of the log file */
+	private static String logFile;
+	
+	/** boolean that enables/disables ERROR logs, 
+	 *  defaults to false (no ERROR logs are logged)
+	 */
+	private boolean enableErrorLog = false;
+	
+	/** boolean that enables/disables WARNING logs, 
+	 *  defaults to false (no WARNING logs are logged)
+	 */
+	private boolean enableWarnLog = false;
+
+	/** boolean that enables/disables SUCCESS logs, 
+	 *  defaults to false (no SUCCESS logs are logged)
+	 */
+	private boolean enableSuccessLog = false;
+
+	/** boolean that enables/disables INFO logs, 
+	 *  defaults to false (no INFO logs are logged)
+	 */
+	private boolean enableInfoLog = false;
+
+	/**
+	 *  boolean to enable/disable testMode while logging
+	 *  If in testMode, logs won't have module or timestamp information attached with it. 
+	 */
+	private boolean inTestMode = false;
+	
+	/**
+	 *  constructor for FileLogger.
+	 *  Protected type since it needs to be only invoked by LoggerManager class
+	 *  @see logger.LoggerManager
```

Comment:
`LoggerManager` not `logger.LoggerManager`

Reply:
yeah, thanks
javadoc is a bit tricky sometimes :)